File: codeabbey/072/TanyaVon18/codeabbey/code/072/FunnyWordsGenerator.java
import java.util.ArrayList;
import java.util.List;

public class FunnyWordsGenerator {

    public static void main(String[] args) {
        // Example usage
        long seed = 0L;
        int wordLength = 6;
        List<Long> randomNumbers = generateRandomNumbers(seed, wordLength);

        String randomWord = generateRandomWord(randomNumbers);
        System.out.println("Random Word: " + randomWord);
    }

    public static List<Long> generateRandomNumbers(long seed, int count) {
        final int A = 445;
        final int C = 700001;
        final int M = 2097152;
        long Xn = seed;
        List<Long> randomNumbers = new ArrayList<>();

        for (int i = 0; i < count; i++) {
            Xn = (A * Xn + C) % M;
            randomNumbers.add(Xn);
        }

        return randomNumbers;
    }

    public static String generateRandomWord(List<Long> randomNumbers) {
        String consonants = "bcdfghjklmnprstvwxz";
        String vowels = "aeiou";
        StringBuilder word = new StringBuilder();

        for (long number : randomNumbers) {
            int index = (int) (word.length() % 2 == 0 ? (number % consonants.length()) : (number % vowels.length()));

            char letter = (char) (word.length() % 2 == 0 ? consonants.charAt(index) : vowels.charAt(index));
            word.append(letter);
        }

        return word.toString();
    }
}

File: codeabbey/072/MahiroEnako/072/FunnyWordsGenerator.kt
fun main() {
    // Example usage
    val seed = 0L
    val wordLength = 6
    val randomNumbers = generateRandomNumbers(seed, wordLength)

    val randomWord = generateRandomWord(randomNumbers)
    println("Random Word: $randomWord")
}

fun generateRandomNumbers(seed: Long, count: Int): List<Long> {
    val A = 445
    val C = 700001
    val M = 2097152
    var Xn = seed
    val randomNumbers = mutableListOf<Long>()

    repeat(count) {
        Xn = (A * Xn + C) % M
        randomNumbers.add(Xn)
    }

    return randomNumbers
}

fun generateRandomWord(randomNumbers: List<Long>): String {
    val consonants = "bcdfghjklmnprstvwxz"
    val vowels = "aeiou"
    val word = StringBuilder()

    for (number in randomNumbers) {
        val index = if (word.length % 2 == 0) {
            (number % consonants.length).toInt()
        } else {
            (number % vowels.length).toInt()
        }

        val letter = if (word.length % 2 == 0) {
            consonants[index]
        } else {
            vowels[index]
        }

        word.append(letter)
    }

    return word.toString()
}
